sdk/cosmos/azure-cosmos-spark_3_2-12/src/test/scala/com/azure/cosmos/spark/PointWriterITest.scala

```diff
@@ -263,6 +276,322 @@ class PointWriterITest extends IntegrationSpec with CosmosClient with AutoCleana
+      StructField("_rid", StringType),
+      StructField("_attachment", StringType)
+    ))
+
+    val columnConfigsMap = new TrieMap[String, CosmosPatchColumnConfig]
+    partialUpdateSchema.fields.foreach(field => {
+      columnConfigsMap += field.name -> CosmosPatchColumnConfig(field.name, CosmosPatchOperationTypes.Set, s"/${field.name}")
+    })
+
+    val pointWriterForPatch = CosmosPatchTestHelper.getPointWriterForPatch(columnConfigsMap, container, partitionKeyDefinition)
+    val patchPartialUpdateItem = CosmosPatchTestHelper.getPatchItemWithSchema(id, partialUpdateSchema, originalItem)
+
+    pointWriterForPatch.scheduleWrite(partitionKey, patchPartialUpdateItem)
+    pointWriterForPatch.flushAndClose()
+
+    val updatedItem: ObjectNode = container.readItem(id, partitionKey, classOf[ObjectNode]).block().getItem
+    updatedItem.get("propInt").asInt() == patchPartialUpdateItem.get("propInt").asInt()
+  }
+
+  "Point Writer" can "patch item with condition" in {
+    val container = getContainer
+    val containerProperties = container.read().block().getProperties
+    val partitionKeyDefinition = containerProperties.getPartitionKeyDefinition
+    val writeConfig = CosmosWriteConfig(
+      ItemWriteStrategy.ItemOverwrite,
+      5,
+      bulkEnabled = false,
+      bulkMaxPendingOperations = Some(900)
+    )
+
+    val pointWriter = new PointWriter(
+      container, partitionKeyDefinition, writeConfig, DiagnosticsConfig(Option.empty, false, None), MockTaskContext.mockTaskContext())
+
+    // First create one item, as patch can only operate on existing items
+    val itemWithFullSchema = CosmosPatchTestHelper.getPatchItemWithFullSchema(UUID.randomUUID().toString)
+    val id = itemWithFullSchema.get("id").textValue()
+    val partitionKey = new PartitionKey(id)
+
+    pointWriter.scheduleWrite(partitionKey, itemWithFullSchema)
+    pointWriter.flushAndClose()
+    // make sure the item exists
+    val originalItem = container.readItem(id, partitionKey, classOf[ObjectNode]).block().getItem()
+
+    // Cosmos patch does not support for system properties
+    // if we send request to patch for them, server is going to return exception with "Invalid patch request: Cannot patch system property"
+    // so the test is to make sure we have skipped these properties and the request can succeed for other properties
+    val partialUpdateSchema = StructType(Seq(
+      StructField("propInt", IntegerType)
+    ))
+
+    val columnConfigsMap = new TrieMap[String, CosmosPatchColumnConfig]
+    partialUpdateSchema.fields.foreach(field => {
+      columnConfigsMap += field.name -> CosmosPatchColumnConfig(field.name, CosmosPatchOperationTypes.Set, s"/${field.name}")
+    })
+
+    val pointWriterForPatch =
+      CosmosPatchTestHelper.getPointWriterForPatch(
+        columnConfigsMap,
+        container,
+        partitionKeyDefinition,
+        Some(s"from c where c.propInt > ${Integer.MAX_VALUE}")) // using a always false condition
```

Comment:
it should have failed with 412/1110, but somehow the exception does not bubble up, will need to figure out why

Reply:
Turns out we do not throw exceptions from PointWriter. Changed to throw exception from PointWriter